File: python/7 kyu/416.py
# https://www.codewars.com/kata/586efc2dcf7be0f217000619/train/python

# Passed

def reverse_slice(s):
    result = []
    string_to_append = ""
    for char in s:
        string_to_append += char
        result.append(string_to_append[::-1])
        
    result = result[::-1]
        
    return result

output = reverse_slice('abcde')
print(output)

File: nodejs/7 kyu/416.js
// https://www.codewars.com/kata/5dd259444228280032b1ed2a/train/javascript

// Passed

function solve(s, g) {
  let result = -1;
  for (let num1 = g; num1 <= Math.ceil(s / 2); num1 += g) {
    let num2 = s - num1;
    if (num2 % g === 0) {
      result = [num1, num2];
      break;
    }
  }

  return result;
}

const output = solve(12, 4);
console.log(output);